cpp/src/arrow/ipc/metadata_internal.cc

```diff
@@ -961,6 +961,58 @@ Status MakeSparseMatrixIndexCSX(FBB& fbb, const SparseIndexType& sparse_index,
   return Status::OK();
 }
 
+Status MakeSparseTensorIndexCSF(FBB& fbb, const SparseCSFIndex& sparse_index,
+                                const std::vector<BufferMetadata>& buffers,
+                                flatbuf::SparseTensorIndex* fb_sparse_index_type,
+                                Offset* fb_sparse_index, size_t* num_buffers) {
+  *fb_sparse_index_type = flatbuf::SparseTensorIndex::SparseTensorIndexCSF;
+  int ndim = sparse_index.axis_order().size();
+
+  // We assume that the value type of indptr tensor is an integer.
+  const auto& indptr_value_type =
+      checked_cast<const IntegerType&>(*sparse_index.indptr()[0]->type());
+  auto indptr_type_offset = flatbuf::CreateInt(fbb, indptr_value_type.bit_width(),
+                                               indptr_value_type.is_signed());
+
+  // We assume that the value type of indices tensor is an integer.
+  const auto& indices_value_type =
+      checked_cast<const IntegerType&>(*sparse_index.indices()[0]->type());
+  auto indices_type_offset = flatbuf::CreateInt(fbb, indices_value_type.bit_width(),
+                                                indices_value_type.is_signed());
+
+  std::vector<const flatbuf::Buffer*> indptr;
+  std::vector<const flatbuf::Buffer*> indices;
+
+  for (int i = 0; i < ndim - 1; ++i) {
+    const BufferMetadata& buffer = buffers[i];
+    flatbuf::Buffer tmp_buffer(buffer.offset, buffer.length);
+    indptr.emplace_back(&tmp_buffer);
```

Comment:
`tmp_buffer` is a temporary variable, so `&tmp_buffer` is invalid just after exiting this block.

Reply:
Changed. The objects now stay in memory after the for loops have completed.